config/management/src/secure_backend.rs

```diff
@@ -127,13 +127,14 @@ impl TryInto<config::SecureBackend> for SecureBackend {
     }
 }
 
+#[macro_export]
 macro_rules! secure_backend {
-    ($struct_name:ident, $field_name:ident, $purpose:literal) => {
+    ($struct_name:ident, $field_name:ident, $purpose:literal, $required:literal) => {
         #[derive(Clone, Debug, StructOpt)]
         pub struct $struct_name {
             #[structopt(long,
                 help = concat!("Backend for ", $purpose),
-                required_unless("config"),
+                required_unless_one(&["config", $required]),
                 long_help = concat!("Backend for ", $purpose, r#"
```

Comment:
You could make this support without the `"ignore"` field by making a second macro path for 

```
($struct_name:ident, $field_name:ident, $purpose:literal) => {
```